### PR TITLE
fix(TUP-19815)Add a problem check for tRunJob

### PR DESCRIPTION
**Pull request by jding-tlnd**
https://jira.talendforge.org/browse/TUP-19815

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/nodes/Node.java [hunk 1]
```diff
@@ -4405,6 +4409,56 @@ private void checkTRunjobwithMRProcess() {
...
+     * version
+     */
+    private void checkMultipleTRunjobVersion() {
+        if (getComponent() != null && "tRunJob".equals(getComponent().getName())) {
+            List<? extends INode> allNodes = process.getGraphicalNodes();
+            IElementParameter thisElement = this.getElementParameter(EParameterName.PROCESS.getName());
+            Object thisVersion = thisElement.getChildParameters().get(EParameterName.PROCESS_TYPE_VERSION.getName()).getValue();
+            String lastVersion = null;
+            try {
+                List<IRepositoryViewObject> all = ProxyRepositoryFactory.getInstance().getAll(ERepositoryObjectType.PROCESS);
```
Comment (cmeng-talend) [inline main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/nodes/Node.java]: Do you mean ProxyRepositoryFactory#getLastVersion?
Please use ProxyRepositoryFactory#getLastVersion(Project project, String id, String folderPath, ERepositoryObjectType type), and I think we need to consider all process types, please refer to ERepositoryObjectType#getAllTypesOfProcess

File: main/plugins/org.talend.designer.core/src/main/java/messages.properties [hunk 2]
```diff
@@ -505,6 +505,7 @@ Node.canNotMultiKindTableOutput=This component can not use the "{0}" and "{1}" t
 Node.ELTDBMap.canNotHaveMultiStartNode=The Link type between "{0}" and "{1}" should be use "{2}".
 Node.checkHasMultiPrejobOrPostJobComponents="{0}" component found in this job,there must be only one totally,Please check it.
 Node.checkTRunjobwithMRProcess=The {1} job is only compatible with independant execution({0}).
+Node.checkMultipleTRunjobVersion=For a job, can't exist multiple child jobs with different versions, please check it.
```
[the remark this answers is not in the document]
Comment (jding-tlnd) [reply main/plugins/org.talend.designer.core/src/main/java/messages.properties]: ok, I'll check with john

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/nodes/Node.java (hunk 1 above)
Comment (jding-tlnd) [reply main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/nodes/Node.java]: Solved

File: main/plugins/org.talend.designer.core/src/main/java/messages.properties (hunk 2 above)
[the remark this answers is not in the document]
Comment (jding-tlnd) [reply main/plugins/org.talend.designer.core/src/main/java/messages.properties]: Solved

**Pull request merged**